src/App.js

```diff
@@ -10,13 +10,35 @@ class App extends Component {
 
     render() {
         return (
-            <div className="App">
-                <input placeholder="city,country code" onChange={this.onCityChange}/>
-                <button onClick={this.onCitySubscribe}>Subscribe</button>
+            <div>
+                <div className="App">
+                    <input placeholder="city,country code" onChange={this.onCityChange}/>
+                    <button onClick={this.onCitySubscribe}>Subscribe</button>
+                    <button onClick={this.onPrint}>Print</button>
+                </div>
+                    {Object.entries(this.state.subscribedCities).map(([city, data]) => this.renderCityData(city, data))}
```

Comment:
Lambda's in JSX are generally frowned upon, so it'd be better to change `renderCityData` to take the `[city, data]` tuple, and use:
```javascript
{Object.entries(this.state.subscribedCities).map(this.renderCityData)}
```

EDIT: In fact, why use `Object.entries` at all? The values of the object contain the `city.name` property, so we don't actually need the keys at all. This simplifies it all to:

```javascript
Object.values(this.state.subscribedCities).map(this.renderCityData)}

...

renderCityData = data => {
  const cityName = data.city.name;
  return data.list.map(datum => ...
}
```

Reply:
Also, this should preferably be in a different component, in a separate file, focused solely on displaying and managing the data